### ISSUE TITLE
Event IDs and wwnames usage

### ISSUE THREAD

**Issue by NSACloud** [bnnm/wwiser#4]
I'm having some trouble understanding how to properly use wwnames.txt. I have a list of event IDs and their corresponding names for Nier Replicant. I'm trying to link event IDs to wem IDs so that every wem uses it's event name.

I've read the guide at https://github.com/bnnm/wwiser-utils/blob/master/doc/NAMES.md, but putting the event names into wwnames.txt doesn't give any results after running `wwiser.pyz nameextract\*.bnk -sl -sm`. I've put all bnks and wems from the game into that folder, but the resulting wwnames list has the event IDs missing their names.

Is there any way to manually specify event IDs and their corresponding names?
wwnames:
![image](https://user-images.githubusercontent.com/46909075/117530971-1bc50880-afae-11eb-982b-2878ce25ff47.png)

Result wwnames:
![image](https://user-images.githubusercontent.com/46909075/117530811-2d59e080-afad-11eb-815d-0707f2e59acc.png)
Event names:
![image](https://user-images.githubusercontent.com/46909075/117530793-15825c80-afad-11eb-8742-b5e60b7fa523.png)

[replicant_EventNames.zip](https://github.com/bnnm/wwiser/files/6445433/replicant_EventNames.zip)

**Comment by bnnm** [comment]
Names in in wwnames.txt are automatically converted to event IDs and used as found (numbers, commas as so on in the file are ignored). bgm_150200_na_event_anxious_d_2mix does become 34091917 so I think wwiser is not reading the txt.

Try putting the .txt together with the .bnk rather than in the program folder (while the .db3 goes near the program, guess the guide isn't not too clear about that). Also check the log isn't writting some related error.

**Comment by NSACloud** [comment]
Thanks, that was the problem. Putting wwnames in the bnk directory fixed it. The new list generated fine and generating txtps worked.

The only problem I have is that all music becomes music_init and doesn't use it's bgm name. 

For example, music_init (music=1178338387)(3551080421=-).txtp should be bgm_110050_na_stage_desert.  The generated wwnames has that event name but it doesn't get used with the txtp's.

Also, is there any built in function of wwiser to export a list of wem ids used in events?
Something like this:
`bgm_110050_na_stage_desert,304185279`
`bgm_110050_na_stage_desert,237302936`
`bgm_110050_na_stage_desert,808764686`

If not, it's not a big deal. I'll just make a python script to read the txtp's.

**Comment by bnnm** [comment]
SFX often use simple events, but the way bgm works in most Wwise games is, they have a single function-like event that plays music (in this case `music_init`), then change the current parameter/song via "setter" events.

`bgm_110050_na_stage_desert` is one of those setters and not a bgm name. The parameter (or rather, variable=value) would update `music` to `na_stage_desert` (1178338387), and `music_init` would react and change current track. 

Setters can't play music by themselves (if `music_init` isn't active nothing plays), thus only .txtp for `music_init` plus different variables can be generated. There can be sub-parameters too, like: `event_name (var1=value1)(var2=value2)(var3=value3)...`. For example one setter could change current track, and other setter set on/off vocals, or language, and so on, so just one setter name doesn't give you the full picture.

Basically, you need to add variables/values names to `wwnames.txt` to get name-like results: `music_init (music=na_stage_desert)(relax=-)`. Note that setter event and parameter don't necessarily match, could simply be `bgm_110050` or `set_desert` that sets `na_stage_desert`, so figuring them out is part of the "fun". I've uploaded a list of most bgm parameters for [Nier here](https://github.com/bnnm/wwiser-utils/blob/master/wwnames/NieR%20Replicant%20(PC).txt), save a few I couldn't identify.

As for generating event names+wem ids there is none, or rather the .txtp are intended to be that. As you see from the explanation above the relation between a .wem and its event(s)+params isn't that obvious.

**Comment by NSACloud** [comment]
That makes much more sense. Thank you for your help, I really appreciate it. I think I can figure out the rest from here so I'll close the issue.
